src/Spectre.libGaClassifier/AllLabelTypesIncludedCondition.cpp

```diff
@@ -0,0 +1,48 @@
+/*
+* AllLabelTypesIncludedCondition.cpp
+* Checks, if Individual has at least specific amount of true values.
+*
+Copyright 2018 Spectre Team
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#include <span.h>
+#include "AllLabelTypesIncludedCondition.h"
+#include "Spectre.libClassifier/NotABinaryLabelException.h"
+#include <set>
+
+namespace spectre::supervised
+{
+AllLabelTypesIncludedCondition::AllLabelTypesIncludedCondition(gsl::span<const supervised::Label> labels, std::unique_ptr<BaseIndividualFeasibilityCondition> condition) :
+    BaseIndividualFeasibilityCondition(std::move(condition)),
+    m_Labels(labels)
+{
+    std::set<Label> labelTypes(labels.begin(), labels.end());
+    m_LabelTypesAmount = labelTypes.size();
```

Comment:
Move this to initializer list (`m_LabelTypesAmount(std::set<Label>(labels.begin(), labels.end()).size)`). This will allow you to make `m_LabelsTypesAmount` `const`.

Reply:
done